include/elements/3d/hexahedron_bspline_element.h

```diff
@@ -0,0 +1,183 @@
+                        const VectorDim& particle_size,
+                        const VectorDim& deformation_gradient) const override;
+
+  //! Compute Jacobian
+  //! \param[in] xi given local coordinates
+  //! \param[in] nodal_coordinates Coordinates of nodes forming the cell
+  //! \param[in] particle_size Particle size
+  //! \param[in] deformation_gradient Deformation gradient
+  //! \retval jacobian Jacobian matrix
+  Eigen::Matrix<double, Tdim, Tdim> jacobian(
+      const Eigen::Matrix<double, 3, 1>& xi,
+      const Eigen::MatrixXd& nodal_coordinates,
+      const Eigen::Matrix<double, 3, 1>& particle_size,
+      const Eigen::Matrix<double, 3, 1>& deformation_gradient) const override;
+
+  //! Compute Jacobian local
+  //! \param[in] xi given local coordinates
+  //! \param[in] nodal_coordinates Coordinates of nodes forming the cell
+  //! \param[in] particle_size Particle size
+  //! \param[in] deformation_gradient Deformation gradient
+  //! \retval jacobian Jacobian matrix
+  Eigen::Matrix<double, Tdim, Tdim> jacobian_local(
+      const Eigen::Matrix<double, 3, 1>& xi,
+      const Eigen::MatrixXd& nodal_coordinates,
+      const Eigen::Matrix<double, 3, 1>& particle_size,
+      const Eigen::Matrix<double, 3, 1>& deformation_gradient) const override;
+  //! Evaluate the B matrix at given local coordinates for a real cell
+  //! \param[in] xi given local coordinates
+  //! \param[in] nodal_coordinates Coordinates of nodes forming the cell
+  //! \param[in] particle_size Particle size
+  //! \param[in] deformation_gradient Deformation gradient
+  //! \retval bmatrix B matrix
+  std::vector<Eigen::MatrixXd> bmatrix(
+      const VectorDim& xi, const Eigen::MatrixXd& nodal_coordinates,
+      const VectorDim& particle_size,
+      const VectorDim& deformation_gradient) const override;
+
+  //! Return the type of shape function
+  mpm::ShapefnType shapefn_type() const override {
+    return mpm::ShapefnType::BSPLINE;
+  }
+
+  //! Return number of shape functions
+  unsigned nfunctions() const override { return nconnectivity_; }
+
+  //! Assign nodal connectivity property for bspline elements
+  //! \param[in] nodal_coordinates Coordinates of nodes forming the cell
+  //! \param[in] nodal_properties Vector determining node type for each
+  //! dimension
+  void initialise_bspline_connectivity_properties(
+      const Eigen::MatrixXd& nodal_coordinates,
+      const std::vector<std::vector<unsigned>>& nodal_properties) override;
+
+  //! Return the degree of shape function
+  mpm::ElementDegree degree() const override {
+    return mpm::ElementDegree::Quadratic;
+  };
+
+ private:
+  //! Compute B-Spline Basis Function using the recursive De Boor's algorithm
+  //! for single direction
+  //! \param[in] point_coord point coordinate in one direction
+  //! \param[in] nodal_coord nodal coordinate in one direction
+  //! dimension
+  //! \param[in] knot_vector Knot vector type associated with direction
```

Comment:
knot_vector should be removed from the argument list (also for 2D)?

Reply:
Yes! I changed the input arguments explanation. Thanks!